### PR TITLE
Adminfacility privileges

### PR DESCRIPTION
**Pull request by dominikl**
# What this PR does

Add methods for getting and setting admin privileges to the gateway.

# Testing this PR

Check that the integration tests (AdminFacilityTest) pass.

# Related reading

https://trello.com/c/FjayBnvM/39-gateway-support-for-roles

### CODE REVIEW

**Comment by dominikl** [comment]
Ready for review, just have to fix some javadoc issues.

**Comment by mtbc** [comment]
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-java/lastCompletedBuild/testngreports/integration.gateway/AdminFacilityTest/ looks good.

File: components/blitz/src/omero/gateway/facility/AdminFacility.java [hunk 1]
```diff
@@ -110,26 +114,72 @@ public GroupData createGroup(SecurityContext ctx, GroupData groupData,
...
+     *            The groups to add the user to.
+     * @param isAdmin
+     *            Pass <code>true</code> if the user is an administrator,
+     *            <code>false</code> otherwise.
+     * @param isGroupOwner
+     *            Pass <code>true</code> if the user is a group owner,
+     *            <code>false</code> otherwise.
      * @return See above.
      * @throws DSOutOfServiceException
      *             If the connection is broken, or not logged in
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/facility/AdminFacility.java]: end with period? (also below)

File: components/blitz/src/omero/gateway/facility/AdminFacility.java [hunk 2]
```diff
@@ -241,6 +297,79 @@ public ExperimenterData lookupExperimenter(SecurityContext ctx, String name)
...
+            ExperimenterData user) throws DSOutOfServiceException,
+            DSAccessException {
+        try {
+            IAdminPrx adm = gateway.getAdminService(ctx);
+            return Utils
+                    .fromEnum(adm.getAdminPrivileges(user.asExperimenter()));
+        } catch (Exception e) {
+            handleException(this, e, "Cannot get admin privileges.");
+        }
+        return Collections.EMPTY_LIST;
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/facility/AdminFacility.java]: you don't get a warning in Eclipse here?

File: components/tools/OmeroJava/test/integration/gateway/AdminFacilityTest.java [hunk 3]
```diff
@@ -61,7 +64,10 @@ public void testCreateExperimenter() throws DSOutOfServiceException, DSAccessExc
         exp.setLastName("User");
         List<GroupData> groups = new ArrayList<GroupData>();
         groups.add(group);
-        exp = adminFacility.createExperimenter(rootCtx, exp, UUID.randomUUID().toString(), "test", groups, false, true);
+        // create a 'light admin' user without any admin privileges
+        exp = adminFacility
+                .createExperimenter(rootCtx, exp, UUID.randomUUID().toString(),
+                        "test", groups, true, true, Collections.EMPTY_LIST);
         Assert.assertTrue(exp.getId()>-1);
     }
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/gateway/AdminFacilityTest.java]: maybe can also assert they don't have any privileges?

File: components/tools/OmeroJava/test/integration/gateway/AdminFacilityTest.java [hunk 4]
```diff
@@ -76,4 +82,17 @@ public void testLookupGroup() throws DSOutOfServiceException, DSAccessException
         GroupData g = adminFacility.lookupGroup(rootCtx, group.getName());
         Assert.assertEquals(group.getId(), g.getId());
     }
+    
+    @Test(dependsOnMethods = { "testCreateExperimenter" })
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/gateway/AdminFacilityTest.java]: add Javadoc describing what the test does

**Comment by mtbc** [comment]
Code looks generally good too.

**Review by mtbc — APPROVED** [review]

**Pull request merged**